Customizable path to executable

DonRyuDragoni opened the pull request:
Change cargo-process to maybe use a custom path for the cargo executable, if one is provided.

This adds a customizable variable, `cargo-process--custom-path-to-bin`, that, if left alone, will (hopefully) make `cargo.el` behave exactly the same as before, and, when set to anything other than an empty string, will adjust the called commands to use the defined path.

I made these changes because Emacs would not find Cargo on my machine (and having to type <kbd>C-u-c-c-b-HOME-~/.cargo/bin/-RET</kbd> every time I needed to compile my code was a bit too much).

Please take this with a grain of salt, as I never coded any elisp before. Everything works in every test that I made, but I'm still unsure.

Commit:
Customizable path to executable

Change cargo-process to maybe use a custom path for the cargo
executable, if one is provided.

kwrooijen commented on cargo-process.el:
```diff
@@ -145,7 +150,12 @@
   (let ((buffer (concat "*Cargo " name "*"))
-        (command (cargo-process--maybe-read-command command))
+        (command (cargo-process--maybe-read-command
+                  (if (string= cargo-process--custom-path-to-bin "")
```
And maybe to clean this up a bit, something like this?

```
 (let* ((buffer (concat "*Cargo " name "*"))
        (path cargo-process--custom-path-to-bin)
        (path (and path (file-name-as-directory path)))
        (command (cargo-process--maybe-read-command (concat path command)))
        (project-root (cargo-process--project-root)))
```

Love it or hate it, it's my preference :)
If Elisp had Clojure's threading macro it would've been much nicer..

kwrooijen commented on cargo-process.el:
```diff
@@ -53,6 +53,11 @@
   :prefix "cargo-process-"
   :group 'cargo)
 
+(defcustom cargo-process--custom-path-to-bin ""
```
Default should be `nil`
(the reason is that you won't have to do an empty string check this way)

Commit:
Code revision: make path nil by default

This means two things:

1. change `cargo-process--custom-path-to-bin` to be nil by default, to
   avoid having to check for an empty string;

2. change let binding in `cargo-process--start` to accomodate 1, and to
   do some cleaning up.

DonRyuDragoni commented:
Sure thing, I'm making the changes now.

kwrooijen commented:
Seems to be working, thanks!
